### PR TITLE
Remove the columns private_key_data and private_key_type from the table gcp_service_account_key Closes #448

### LINKED ISSUE #448: Remove the columns private_key_data and private_key_type from the table gcp_service_account_key & fix public_key_data to render data
`**Describe` the bug**
For security purposes, private_key_data and private_key_type should not be expose.

**Steampipe version (`steampipe -v`)**
Example: v0.3.0

**Plugin version (`steampipe plugin list`)**
Example: v0.5.0

**To reproduce**
Steps to reproduce the behavior (please include relevant code and/or commands).

**Expected behavior**

**Additional context**

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/gcp_service_account_key []

PRETEST: tests/gcp_service_account_key

TEST: tests/gcp_service_account_key
Running terraform
data.google_client_config.current: Reading...
data.google_client_config.current: Read complete after 0s [id=projects/"parker-aaa"/regions/"us-east1"/zones/"us-east1-b"]
data.null_data_source.resource: Reading...
data.null_data_source.resource: Read complete after 0s [id=static]

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # google_service_account.named_test_resource will be created
  + resource "google_service_account" "named_test_resource" {
      + account_id   = "turbottest75749"
      + description  = "Test service account to verify the table"
      + disabled     = false
      + display_name = "turbottest75749"
      + email        = (known after apply)
      + id           = (known after apply)
      + member       = (known after apply)
      + name         = (known after apply)
      + project      = (known after apply)
      + unique_id    = (known after apply)
    }

  # google_service_account_key.named_test_resource will be created
  + resource "google_service_account_key" "named_test_resource" {
      + id                 = (known after apply)
      + key_algorithm      = "KEY_ALG_RSA_2048"
      + name               = (known after apply)
      + private_key        = (sensitive value)
      + private_key_type   = "TYPE_GOOGLE_CREDENTIALS_FILE"
      + public_key         = (known after apply)
      + public_key_type    = "TYPE_X509_PEM_FILE"
      + service_account_id = (known after apply)
      + valid_after        = (known after apply)
      + valid_before       = (known after apply)
    }

Plan: 2 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + name                 = (known after apply)
  + project_id           = "parker-aaa"
  + resource_aka         = (known after apply)
  + resource_id          = (known after apply)
  + service_account_name = (known after apply)
google_service_account.named_test_resource: Creating...
google_service_account.named_test_resource: Creation complete after 3s [id=projects/parker-aaa/serviceAccounts/turbottest75749@parker-aaa.iam.gserviceaccount.com]
google_service_account_key.named_test_resource: Creating...
google_service_account_key.named_test_resource: Creation complete after 2s [id=projects/parker-aaa/serviceAccounts/turbottest75749@parker-aaa.iam.gserviceaccount.com/keys/97648cc36d8cfdb70040f0b43da067e1d1d719ba]

Warning: Deprecated

  with data.null_data_source.resource,
  on variables.tf line 33, in data "null_data_source" "resource":
  33: data "null_data_source" "resource" {

The null_data_source was historically used to construct intermediate values
to re-use elsewhere in configuration, the same can now be achieved using
locals

(and one more similar warning elsewhere)

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

name = "97648cc36d8cfdb70040f0b43da067e1d1d719ba"
project_id = "parker-aaa"
resource_aka = "gcp://iam.googleapis.com/projects/parker-aaa/serviceAccounts/turbottest75749@parker-aaa.iam.gserviceaccount.com/keys/97648cc36d8cfdb70040f0b43da067e1d1d719ba"
resource_id = "projects/parker-aaa/serviceAccounts/turbottest75749@parker-aaa.iam.gserviceaccount.com/keys/97648cc36d8cfdb70040f0b43da067e1d1d719ba"
service_account_name = "turbottest75749@parker-aaa.iam.gserviceaccount.com"

Running SQL query: test-get-query.sql
[
  {
    "key_algorithm": "KEY_ALG_RSA_2048",
    "key_origin": "GOOGLE_PROVIDED",
    "key_type": "USER_MANAGED",
    "location": "global",
    "name": "97648cc36d8cfdb70040f0b43da067e1d1d719ba",
    "project": "parker-aaa",
    "service_account_name": "turbottest75749@parker-aaa.iam.gserviceaccount.com"
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "key_algorithm": "KEY_ALG_RSA_2048",
    "key_origin": "GOOGLE_PROVIDED",
    "key_type": "USER_MANAGED",
    "name": "97648cc36d8cfdb70040f0b43da067e1d1d719ba",
    "service_account_name": "turbottest75749@parker-aaa.iam.gserviceaccount.com"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "key_type": "USER_MANAGED",
    "name": "97648cc36d8cfdb70040f0b43da067e1d1d719ba"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "gcp://iam.googleapis.com/projects/parker-aaa/serviceAccounts/turbottest75749@parker-aaa.iam.gserviceaccount.com/keys/97648cc36d8cfdb70040f0b43da067e1d1d719ba"
    ],
    "title": "97648cc36d8cfdb70040f0b43da067e1d1d719ba"
  }
]
✔ PASSED

POSTTEST: tests/gcp_service_account_key

TEARDOWN: tests/gcp_service_account_key

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select * from gcp_service_account_key where service_account_name like '%rk-deleteme-testing-dasdafffg%'
+------------------------------------------+-------------------------------------------------------------------+----------------+------------------+-----------------+------------------------------------->
| name                                     | service_account_name                                              | key_type       | key_algorithm    | key_origin      | public_key_data                     >
+------------------------------------------+-------------------------------------------------------------------+----------------+------------------+-----------------+------------------------------------->
| 4049d7d1853d4fa256c5816387cd0dd2c6f57b1d | rk-deleteme-testing-dasdafffgta@myproject2.iam.gserviceaccount.com | SYSTEM_MANAGED | KEY_ALG_RSA_2048 | GOOGLE_PROVIDED | LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0t>
|                                          |                                                                   |                |                  |                 | ZjhFQkFNQ0I0QXdGZ1lEVlIwbEFRSC9CQXd3>
| bfd36b4c261012e79067fc3dd7b53b2e572efbfc | rk-deleteme-testing-dasdafffgta@myproject2.iam.gserviceaccount.com | USER_MANAGED   | KEY_ALG_RSA_2048 | GOOGLE_PROVIDED | LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0t>
|                                          |                                                                   |                |                  |                 | KzlWSEVaSjlXeGhnR2xKVnJnUU5vSkk4eHpi>
| 055f382e2b1873a6c2a660d5fd5a0f37720a4b52 | rk-deleteme-testing-dasdafffgta@myproject2.iam.gserviceaccount.com | USER_MANAGED   | KEY_ALG_RSA_2048 | USER_PROVIDED   | LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0t>
|                                          |                                                                   |                |                  |                 | ZVIwcGJKV2ZDME5ER25RL1dsMmx6UURtbTdD>
+------------------------------------------+-------------------------------------------------------------------+----------------+------------------+-----------------+------------------------------------->
```
</details>

### CODE REVIEW

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@ParthaI Please see review comment, thanks!

File: gcp/table_gcp_service_account_key.go [hunk 1]
```diff
@@ -28,53 +28,45 @@ func tableGcpServiceAccountKey(_ context.Context) *plugin.Table {
...
+				Description: "Specifies the origin of the key.",
 				Type:        proto.ColumnType_STRING,
 			},
 			{
 				Name:        "public_key_data",
-				Description: "Specifies the public key data",
+				Description: "Specifies the public key data.",
 				Type:        proto.ColumnType_STRING,
+				Hydrate:     getGcpServiceAccountKey,
+				Transform:   transform.FromField("PublicKeyData"),
```
[the remark this answers is not in the document]
Comment (ParthaI) [reply gcp/table_gcp_service_account_key.go:60]: @cbruno10 Actually we don't need this transform function, during testing I was getting null value for this column, after adding this transform function we are also not getting the date, I will remove the this transform function.

File: gcp/table_gcp_service_account_key.go [hunk 2]
```diff
@@ -161,7 +162,9 @@ func getGcpServiceAccountKey(ctx context.Context, d *plugin.QueryData, h *plugin
 
 	keyName := "projects/" + project + "/serviceAccounts/" + serviceAccountName + "/keys/" + name
 
-	op, err := service.Projects.ServiceAccounts.Keys.Get(keyName).Do()
+	queryParameter := googleapi.QueryParameter("publicKeyType", "TYPE_RAW_PUBLIC_KEY", "TYPE_X509_PEM_FILE")
```
Comment (rajlearner17) [inline gcp/table_gcp_service_account_key.go:165]: Can we add any doc/api references as comment or something? Why are we passing these parameters? 

image

**Comment by rajlearner17** [comment]
@ParthaI Thanks now the `public_key_data` is rendered
I would suggest adding an example to highlight this change

```
select 
  name, 
  key_type,
  key_origin
  public_key_data 
from 
  gcp_service_account_key 
where service_account_name = 'deleteme-testing-pubkeydata@parker-aaa.iam.gserviceaccount.com' 
```

**Comment by cbruno10** [comment]
@ParthaI can you please share an example of raw and pem format for the public key data? Also, what does GCP default to?

**Comment by ParthaI** [comment]
> @ParthaI can you please share an example of raw and pem format for the public key data? Also, what does GCP default to?

@cbruno10 Below are the examples with two formats.

**We supplied to generate Key.**

```
-----BEGIN CERTIFICATE-----
MIICnjCCAYYCCQDP9KqsK9Ry7jANBgkqhkiG9w0BAQsFADARMQ8wDQYDVQQDDAZ1
bnVzZWQwHhcNMjMwNTE3MDgzNzA5WhcNMjQwNTE2MDgzNzA5WjARMQ8wDQYDVQQD
DAZ1bnVzZWQwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQC0sOzWCa80
FT1Kjh4SwiCQhi2sknRlxPoQkXc+qXwZ8eYL5galXlqEK2DsIeECazZGi+yqEUJk
DwulPMa90SdEK47dpbC0dD4GcaB63pdyzRcPZNt3fWB6Z1vUZELT4MFhJoCN+4fA
XIZPWoy9BqCBmRjsqwk8dUXM0yyJVuHaFgN4PPxvC2UYn2gqL00uD+baafYP+rch
QJWY4qTTarQ4hzwtbnje7S+LntbcmNQQZEBdBKWnc9+sASzfBfkSlMp9PNGpSNLn
TaNT0WMQGZTy+wQ5vBL9J7Yfq52LCUrXpdVcFwkn9S0UT10L0DlZHyct5RP+h/RJ
SNO4ER5kQi7zAgMBAAEwDQYJKoZIhvcNAQELBQADggEBAF20QrDtNsX4vP6leZlO
5fDrCnlGEq+XacFkyLDWudl1v/wGOGoUb9WW9TPnmrVJL2UUmdWgDk5TholEt0v0
4d3RXqBGLeACwRSu6YWW735IENgVOme8RCaSu8Im/R75NFeOboy2uj9m7OdePJR6
PZBHyTJXGb6bAPRrxHhtDkM+XeR0pbJWfC0NDGnQ/Wl2lzQDmm7CiCeZORlNnGyw
i7i8WAde9WePF6lqhGIuIZh8DDwqchmjsjHlA8/WeLnu2iUtlwaJlpNzZ6j//IA3
5aY11SGxYEmQPzGSwSFnBXBaJgdkcFuum8wwuaiGOZM0wgxN3rdvysFJVokolvl0
GvM=
-----END CERTIFICATE-----
```

**API  returns in this PEM format:**

```
 "publicKeyData": "LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUNuakNDQVlZQ0NRRFA5S3FzSzlSeTdqQU5CZ2txaGtpRzl3MEJBUXNGQURBUk1ROHdEUVlEVlFRRERBWjEKYm5WelpXUXdIaGNOTWpNd05URTNNRGd6TnpBNVdoY05NalF3TlRFMk1EZ3pOekE1V2pBUk1ROHdEUVlEVlFRRApEQVoxYm5WelpXUXdnZ0VpTUEwR0NTcUdTSWIzRFFFQkFRVUFBNElCRHdBd2dnRUtBb0lCQVFDMHNPeldDYTgwCkZUMUtqaDRTd2lDUWhpMnNrblJseFBvUWtYYytxWHdaOGVZTDVnYWxYbHFFSzJEc0llRUNhelpHaSt5cUVVSmsKRHd1bFBNYTkwU2RFSzQ3ZHBiQzBkRDRHY2FCNjNwZHl6UmNQWk50M2ZXQjZaMXZVWkVMVDRNRmhKb0NOKzRmQQpYSVpQV295OUJxQ0JtUmpzcXdrOGRVWE0weXlKVnVIYUZnTjRQUHh2QzJVWW4yZ3FMMDB1RCtiYWFmWVArcmNoClFKV1k0cVRUYXJRNGh6d3RibmplN1MrTG50YmNtTlFRWkVCZEJLV25jOStzQVN6ZkJma1NsTXA5UE5HcFNOTG4KVGFOVDBXTVFHWlR5K3dRNXZCTDlKN1lmcTUyTENVclhwZFZjRndrbjlTMFVUMTBMMERsWkh5Y3Q1UlAraC9SSgpTTk80RVI1a1FpN3pBZ01CQUFFd0RRWUpLb1pJaHZjTkFRRUxCUUFEZ2dFQkFGMjBRckR0TnNYNHZQNmxlWmxPCjVmRHJDbmxHRXErWGFjRmt5TERXdWRsMXYvd0dPR29VYjlXVzlUUG5tclZKTDJVVW1kV2dEazVUaG9sRXQwdjAKNGQzUlhxQkdMZUFDd1JTdTZZV1c3MzVJRU5nVk9tZThSQ2FTdThJbS9SNzVORmVPYm95MnVqOW03T2RlUEpSNgpQWkJIeVRKWEdiNmJBUFJyeEhodERrTStYZVIwcGJKV2ZDME5ER25RL1dsMmx6UURtbTdDaUNlWk9SbE5uR3l3Cmk3aThXQWRlOVdlUEY2bHFoR0l1SVpoOEREd3FjaG1qc2pIbEE4L1dlTG51MmlVdGx3YUpscE56WjZqLy9JQTMKNWFZMTFTR3hZRW1RUHpHU3dTRm5CWEJhSmdka2NGdXVtOHd3dWFpR09aTTB3Z3hOM3JkdnlzRkpWb2tvbHZsMApHdk09Ci0tLS0tRU5EIENFUlRJRklDQVRFLS0tLS0=",
```
**Decoding the PEM format:**

```
-----BEGIN CERTIFICATE-----
MIICnjCCAYYCCQDP9KqsK9Ry7jANBgkqhkiG9w0BAQsFADARMQ8wDQYDVQQDDAZ1
bnVzZWQwHhcNMjMwNTE3MDgzNzA5WhcNMjQwNTE2MDgzNzA5WjARMQ8wDQYDVQQD
DAZ1bnVzZWQwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQC0sOzWCa80
FT1Kjh4SwiCQhi2sknRlxPoQkXc+qXwZ8eYL5galXlqEK2DsIeECazZGi+yqEUJk
DwulPMa90SdEK47dpbC0dD4GcaB63pdyzRcPZNt3fWB6Z1vUZELT4MFhJoCN+4fA
XIZPWoy9BqCBmRjsqwk8dUXM0yyJVuHaFgN4PPxvC2UYn2gqL00uD+baafYP+rch
QJWY4qTTarQ4hzwtbnje7S+LntbcmNQQZEBdBKWnc9+sASzfBfkSlMp9PNGpSNLn
TaNT0WMQGZTy+wQ5vBL9J7Yfq52LCUrXpdVcFwkn9S0UT10L0DlZHyct5RP+h/RJ
SNO4ER5kQi7zAgMBAAEwDQYJKoZIhvcNAQELBQADggEBAF20QrDtNsX4vP6leZlO
5fDrCnlGEq+XacFkyLDWudl1v/wGOGoUb9WW9TPnmrVJL2UUmdWgDk5TholEt0v0
4d3RXqBGLeACwRSu6YWW735IENgVOme8RCaSu8Im/R75NFeOboy2uj9m7OdePJR6
PZBHyTJXGb6bAPRrxHhtDkM+XeR0pbJWfC0NDGnQ/Wl2lzQDmm7CiCeZORlNnGyw
i7i8WAde9WePF6lqhGIuIZh8DDwqchmjsjHlA8/WeLnu2iUtlwaJlpNzZ6j//IA3
5aY11SGxYEmQPzGSwSFnBXBaJgdkcFuum8wwuaiGOZM0wgxN3rdvysFJVokolvl0
GvM=
-----END CERTIFICATE-----
```

**This is RAW format:**

```
  "publicKeyData": "LS0tLS1CRUdJTiBQVUJMSUMgS0VZLS0tLS0KTUlJQklqQU5CZ2txaGtpRzl3MEJBUUVGQUFPQ0FROEFNSUlCQ2dLQ0FRRUF0TERzMWdtdk5CVTlTbzRlRXNJZwprSVl0ckpKMFpjVDZFSkYzUHFsOEdmSG1DK1lHcFY1YWhDdGc3Q0hoQW1zMlJvdnNxaEZDWkE4THBUekd2ZEVuClJDdU8zYVd3dEhRK0JuR2dldDZYY3MwWEQyVGJkMzFnZW1kYjFHUkMwK0RCWVNhQWpmdUh3RnlHVDFxTXZRYWcKZ1prWTdLc0pQSFZGek5Nc2lWYmgyaFlEZUR6OGJ3dGxHSjlvS2k5TkxnL20ybW4yRC9xM0lVQ1ZtT0trMDJxMApPSWM4TFc1NDN1MHZpNTdXM0pqVUVHUkFYUVNscDNQZnJBRXMzd1g1RXBUS2ZUelJxVWpTNTAyalU5RmpFQm1VCjh2c0VPYndTL1NlMkg2dWRpd2xLMTZYVlhCY0pKL1V0RkU5ZEM5QTVXUjhuTGVVVC9vZjBTVWpUdUJFZVpFSXUKOHdJREFRQUIKLS0tLS1FTkQgUFVCTElDIEtFWS0tLS0tCg==",
```

By decoding the RAW format, we are not getting the actual format we provided during the resource creation.

During the resource creation, we must have to provide the data with PEM format.

IMO it will be better if we display the data in PEM format.

**Comment by cbruno10** [comment]
Let's add two columns then, `public_key_data_pem` (which has Base64 decoded data) and `public_key_data_raw`. The `public_key_data_pem` can hydrate from the Get hydrate call, and we can add another hydrate call to get the raw public key data.

**Comment by ParthaI** [comment]
Got it @cbruno10, thanks for the clarification. 👍

**Review by cbruno10 — APPROVED** [review]
